lib/presentation/search/search_page.dart

```diff
@@ -10,105 +17,115 @@ import 'widgets/current_location_widget.dart';
+                onPressed: () => AppRouter().goBack(context),
+                icon: Image.asset(
+                  Images.icClose,
+                  width: 32,
+                  height: 32,
+                ),
               ),
-            ),
-            const SizedBox(height: 24),
-            Text(
-              Strings.getWeather,
-              style: Theme.of(context).textTheme.headlineMedium,
-            ),
-            const SizedBox(height: 32),
-            SearchTextField(
-              controller: _controller,
-              focusNode: _focusNode,
-              onTap: () => setState(() {}),
-              onCloseTap: () => setState(() {
-                _controller.clear();
-                _focusNode.unfocus();
-              }),
-            ),
-            const SizedBox(height: 32),
-            if (!_focusNode.hasFocus) ...[
-              CurrentLocationWidget(locationName: widget.chosenCity.value),
               const SizedBox(height: 24),
-              Divider(
-                color: AppColors.deepPurple.withOpacity(0.1),
-                height: 1,
+              Text(
+                Strings.getWeather,
+                style: Theme.of(context).textTheme.headlineMedium,
               ),
-              const SizedBox(height: 24),
-              Expanded(
-                child: PastSearchBlock(
-                  pastSearchCities: widget.pastSearchCities,
-                  onClearAllTap: () {},
+              const SizedBox(height: 32),
+              SearchTextField(
+                controller: controller,
+                focusNode: focusNode,
+                onCloseTap: () {
+                  controller.clear();
+                  focusNode.unfocus();
+                },
+                onEditingCompleted: (String value) => onEditingComplete(
+                  cityName: value,
                 ),
               ),
-            ] else ...[
-              Expanded(
-                child: ListView.separated(
-                  physics: const NeverScrollableScrollPhysics(),
-                  padding: EdgeInsets.zero,
-                  itemCount: testCities.length,
-                  itemBuilder: (BuildContext context, int index) =>
-                      CitiesListItemWidget(
-                    cityItem: testCities[index],
-                    onTap: () => onCityItemTap(testCities[index].name),
+              const SizedBox(height: 32),
+              if (!focusNode.hasFocus) ...[
+                CurrentLocationWidget(locationName: currentCity?.name ?? ''),
+                const SizedBox(height: 24),
+                Divider(
+                  color: AppColors.deepPurple.withOpacity(0.1),
+                  height: 1,
+                ),
+                const SizedBox(height: 24),
+                Expanded(
+                  child: AppStateConnector<List<String>>(
+                    converter: (state) =>
+                        state.pastSearchCities.pastSearchCities.toList(),
```

Comment:
```suggestion
                        state.pastSearchCities.pastSearchCities,
```

можно builtList использовать на UI, т.к. иначе весь смысл builtList теряется, т.к. можно мутировать список, который достали из стейта